pkg/platform/api/buildplanner/model/buildplan.go

```diff
@@ -282,31 +290,29 @@ func (b *BuildPlanByCommit) Build() (*Build, error) {
 				}
 			}
 		}
-		return nil, &BuildPlannerError{
+		return &BuildPlannerError{
 			ValidationErrors: errs,
 			IsTransient:      isTransient,
 		}
 	}
 
-	return b.Commit.Build, nil
+	return locale.NewInputError("err_buildplanner_build", "Encountered error processing build response")
 }
 
-func (b *BuildPlanByCommit) CommitID() (strfmt.UUID, error) {
-	if b.Commit == nil {
-		return "", errs.New("BuildPlanByCommit.CommitID: Commit is nil")
+func ProcessProjectError(project *Project, fallbackMessage string) error {
+	if project.Error == nil {
+		return errs.New(fallbackMessage)
 	}
 
-	if b.Commit.Error != nil {
-		if b.Commit.Error.Message != "" {
-			return "", errs.New("BuildPlanByCommit.CommitID: Could not get commit ID, API returned commit error message: %s", b.Commit.Message)
-		}
-		return "", errs.New("BuildPlanByCommit.CommitID: Could not retrieve commit")
+	if project.Type == NotFoundErrorType {
+		return locale.NewInputError("err_buildplanner_project_not_found", "Unable to find project, recieved message: {{.V0}}", project.Message)
 	}
 
-	return b.Commit.CommitID, nil
+	return errs.New(fallbackMessage)
```

Comment:
We don't want to silence this one (input error)?

Reply:
If we have to use the fallback message we've encountered an error type that we aren't handling so we should update these functions.